KELOMPOK 6
1. Silvia Zahro (11210940000028) 
2. Salsa Ilhammita (11210940000018)
3. Rizqi Maulidi ( 11210940000022 )
4. Auliyatunnisa' Febrianingrum (11210940000010) 
5. Chelsea Fatihah Rahma (11210940000035) 
6. Ferdian Adi Pratama (11210940000038) 
7. Fadli Naufal Ardianto (11210940000005)

Permasalahan:
1. Apakah ada dosen yang cenderung memberikan nilai cenderung lebih tinggi (atau lebih rendah) dibandingkan dosen lainnya?
2. Menurut data yang ada, apakah terdapat perbedaan dalam pemberian nilai antara dosen dengan Pendidikan Terakhir S3 dan S2?
3. Apakah terdapat perbedaan performa akademik antara mahasiswa yang berasal dari Madrasah/SMK dan SMU?
4. Apakah ada perbedaan yang nampak di data terhadap kemampuan akademik antara mahasiswa dan mahasiswi?
5. Apakah ada perbedaan prestasi akademik yang nampak di data antar angkatan mahasiswa?
6. Apakah ada permasalahan di data atau informasi lain yang anda temukan di data?

In [ ]:
import pandas as pd
mengajar = pd.read_csv('/content/mengajar.csv')
dosen = pd.read_csv('/content/tb_dosen.csv')
kelas = pd.read_csv('/content/tb_kelas.csv')
krs = pd.read_csv('/content/tb_krs.csv')
mahasiswa = pd.read_csv('/content/tb_mahasiswa.csv')
matkul = pd.read_csv('/content/tb_mata_kuliah.csv')
nilai = pd.read_csv('/content/tb_nilai.csv')

nilai_dosen = nilai.groupby(nilai['NIDN']).mean()[['nilai']]
nilai_dosen = nilai_dosen.rename({'nilai': 'mean_nilai'}, axis=1) 
nilai_dosen = pd.merge(dosen, nilai_dosen, left_on='NIDN', right_index=True)
nilai_dosen = nilai_dosen[['NIDN', 'nama', 'mean_nilai']]
sb = nilai.groupby(nilai['NIDN']).std()[['nilai']]
sb = sb.rename({'nilai': 'std_nilai'}, axis=1) 
nilai_dosen = nilai_dosen.set_index(['NIDN']) 
nilai_dosen = pd.concat([nilai_dosen, sb], axis=1)
nilai_dosen


nama  mean_nilai  std_nilai
NIDN                                      
123    Andri Basuki   85.750000   7.741355
234        zulfikar   80.428571  15.883504
345   Annisa Fitria   60.090909  16.220077
787            yudi   70.571429  13.010984

Interpretasi : 
- mean_nilai : rata-rata dari keseluruhan nilai yang ada pada data
- std_nilai : simpangan baku
- kita dapat mengetahui kecenderungan dosen dalam memberikan nilai kepada mahasiswa adalah dari tinggi rendahnya mean_nilai
- pada data di atas, mean_nilai terendah didapat oleh bu Annisa Fitria yaitu 60.090909 dan standar deviasi 16.220077. maka dapat diasumsikan bahwa bu Annisa Fitria merupakan dosen yang cenderung memberikan nilai rendah
sedangkan, mean_nilai tertinggi didapat oleh pak Andri Basuki yaitu 85.750000 dan standar deviasi 7.741355 . maka dapat diasumsikan bahwa Pak Andri Basuki merupakan dosen yang cenderung memberikan nilai tinggi

In [ ]:
#nomor 2
dosen = pd.read_csv('/content/tb_dosen.csv')
dosen = dosen.set_index(['NIDN'])
dosen.drop("nama", axis=1, inplace=True)
dosen.drop("email", axis=1, inplace=True)
dosen.drop("Telephone", axis=1, inplace=True)
dosen.drop("Jenis_Kelamin", axis=1, inplace=True)
dosen

Pendidikan_Terakhir
NIDN                    
123                   S3
234                   S3
787                   S2
345                   S2

In [ ]:
dosen = pd.concat([nilai_dosen, dosen], axis=1)
dosen

nama  mean_nilai  std_nilai Pendidikan_Terakhir
NIDN                                                          
123    Andri Basuki   85.750000   7.741355                  S3
234        zulfikar   80.428571  15.883504                  S3
345   Annisa Fitria   60.090909  16.220077                  S2
787            yudi   70.571429  13.010984                  S2

Interpretasi : 
- mean_nilai : rata-rata dari keseluruhan nilai yang ada pada data
- std_nilai : simpangan baku
- dari data di atas kita peroleh bahwa ada perbedaan pemberian nilai antara dosen lulusan S3 dan S2
- pada data di atas terlihat bahwa rata-rata pemberian nilai dosen S3 lebih tinggi daripada dosen S2
- rata-rata pemberian nilai oleh dosen S3 yaitu 85.750000 dan 80.428571 sedangkan rata-rata pemberian nilai oleh dosen S2 yaitu 60.090909 dan 70.571429 
- demikian sehingga kita dapat asumsikan bahwa rata-rata pemberian nilai oleh dosen S3 lebih tinggi dibandingkan rata-rata pemberian nilai oleh dosen S2

In [ ]:
#nomor 3
nilai_mahasiswa = nilai.groupby(mahasiswa['Asal_Sekolah']).mean()[['nilai']]
nilai_mahasiswa = nilai_mahasiswa.rename({'nilai': 'mean_nilai'}, axis=1)
nilai_mahasiswa

mean_nilai
Asal_Sekolah            
MA             76.250000
SMK            89.400000
SMU            71.333333

Interpretasi : 
- mean_nilai : rata-rata dari keseluruhan nilai yang ada pada data
- dari data di atas, kita peroleh bahwa adanya perbedaan perfoma antara SMU/MA/SMK. bisa dilihat dari mean_nilai yang ada. 
- pada data di atas, mean_nilai terendah berasal dari SMU dengan mean_nilai 71.333333, sedangkan mean_nilai tertinggi berasal dari SMK dengan mean_nilai 89.400000
- lalu adapun dari MA dengan mean_nilai 76.2500000
- demikian sehingga dapat diasumsikan bahwa SMU memiliki perfoma akademik yang rendah dibandingkan MA dan SMK begitupun dengan MA yang performa akademiknya tidak lebih jauh dari SMU. sedangkan SMK memiliki performa akademik yang bagus ketimbang SMU dan MA 

In [ ]:
#nomor 4
nilai_gender = nilai.groupby(mahasiswa['Gender']).mean()[['nilai']] 
nilai_gender = nilai_gender.rename({'Gender': 'nilai'}, axis=1)
nilai_gender

nilai
Gender           
Pria    81.222222
Wanita  74.833333

Interpretasi : 
- gender : jenis kelamin yang dimiliki oleh setiap mahasiswa/i
- nilai : nilai rata-rata mahasiswa yang ada pada data
- dari data di atas kita peroleh bahwa ada perbedaan kemapuan akademik antara pria dan wanita
- pada data di atas terlihat bahwa nilai rata rata mahasiswa lebih tinggi daripada mahasiswi
-nilai rata-rata mahasiswa yaitu 81.222222 sedangkan nilai rata-rata mahasiswi yaitu 74.833333 
- demikian sehingga dapat diasumsikan bahwa nilai rata-rata mahasiswa lebih tinggi daripada nilai rata-rata mahasiswi

In [ ]:
#nomor 5
nilai_angkatan = nilai.groupby(mahasiswa['Angkatan']).mean()[['nilai']] 
nilai_angkatan = nilai_angkatan.rename({'Angkatan': 'nilai'}, axis=1)
nilai_angkatan

nilai
Angkatan           
2019.0    85.750000
2020.0    70.571429

Interpretasi : 
- dari data di atas kita peroleh bahwa ada perbedaan pretasi akademik antara angkatan mahasiswa
- pada data di atas terlihat bahwa nilai rata rata angkatan 2019 lebih tinggi daripada angkatan 2020
- nilai rata rata angkatan mahasiswa 2019 yaitu 85.750000 sedangkan nilai rata rata angkatan mahasiswa 2020 yaitu 70.571429
- demikian sehingga dapat diasumsikan bahwa nilai rata-rata angkatan mahasiswa 2019 lebih tinggi dari nilai rata-rata angkatan mahasiswa 2020

In [ ]:
#nomor 6


Jawaban : 
- informasi lain pada data yaitu terlalu banyak variable yang tidak diperlukan sehingga kita harus mengelolanya terlebih dahulu sebelum dilakukannya analisis
- terdapat satu variable yaitu rand yang tidak memiliki nilai sehingga tidak diperlukan

Kesimpulan : 
1. Apakah ada dosen yang cenderung memberikan nilai cenderung lebih tinggi (atau lebih rendah) dibandingkan dosen lainnya? 
- pada data di atas, mean_nilai terendah didapat oleh bu Annisa Fitria yaitu 60.090909 dan standar deviasi 16.220077. maka dapat diasumsikan bahwa bu Annisa Fitria merupakan dosen yang cenderung memberikan nilai rendah
- sedangkan, mean_nilai tertinggi didapat oleh pak Andri Basuki yaitu 85.750000 dan standar deviasi 7.741355. maka dapat diasumsikan bahwa Pak Andri Basuki merupakan dosen yang cenderung memberikan nilai tinggi
2. Menurut data yang ada, apakah terdapat perbedaan dalam pemberian nilai antara dosen dengan Pendidikan Terakhir S3 dan S2? 
- rata-rata pemberian nilai oleh dosen S3 lebih tinggi dibandingkan rata-rata pemberian nilai oleh dosen S2
3. Apakah terdapat perbedaan performa akademik antara mahasiswa yang berasal dari Madrasah/SMK dan SMU? 
- SMU memiliki perfoma akademik yang rendah dibandingkan MA dan SMK begitupun dengan MA yang performa akademiknya tidak lebih jauh dari SMU. sedangkan SMK memiliki performa akademik yang bagus ketimbang SMU dan MA 
4. Apakah ada perbedaan yang nampak di data terhadap kemampuan akademik antara mahasiswa dan mahasiswi? 
- nilai rata-rata mahasiswa lebih tinggi daripada nilai rata-rata mahasiswi
5. Apakah ada perbedaan prestasi akademik yang nampak di data antar angkatan mahasiswa? 
- nilai rata-rata angkatan mahasiswa 2019 lebih tinggi dari nilai rata-rata angkatan mahasiswa 2020
6. Apakah ada permasalahan di data atau informasi lain yang anda temukan di data?
- informasi lain pada data yaitu terlalu banyak variable yang tidak diperlukan sehingga kita harus mengelolanya terlebih dahulu sebelum dilakukannya analisis
- terdapat satu variable yaitu rand yang tidak memiliki nilai sehingga tidak diperlukan